packages/template-retail-react-app/app/pages/product-list/partials/page-designer-promotional-banner.jsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import {usePage} from '@salesforce/commerce-sdk-react'
+
+// Components
+import {Box} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {Page} from '@salesforce/commerce-sdk-react/components'
+
+// Page Designer Components
+import {ImageWithText} from '@salesforce/retail-react-app/app/page-designer/assets'
+import {MobileGrid1r1c} from '@salesforce/retail-react-app/app/page-designer/layouts'
+
+const PageDesignerPromotionalBanner = () => {
+    const PROMO_BANNER_DESKTOP_PAGE_ID = 'instagram-promo-banner-desktop'
+    const PROMO_BANNER_MOBILE_PAGE_ID = 'instagram-promo-banner-mobile'
+    const PAGEDESIGNER_TO_COMPONENT = {
+        'commerce_assets.productListTile': ImageWithText,
+        'commerce_layouts.mobileGrid1r1c': MobileGrid1r1c
+    }
+    const {data: promoBannerDesktop, error: pageErrorDesktop} = usePage({
+        parameters: {pageId: PROMO_BANNER_DESKTOP_PAGE_ID}
+    })
+    const {data: promoBannerMobile, error: pageErrorMobile} = usePage({
+        parameters: {pageId: PROMO_BANNER_MOBILE_PAGE_ID}
+    })
+
+    return (
+        <Box>
+            {promoBannerDesktop && !pageErrorDesktop && (
+                <Box display={{base: 'none', md: 'block'}}>
+                    <Page
+                        page={promoBannerDesktop}
+                        components={PAGEDESIGNER_TO_COMPONENT}
+                        data-testid={'sf-promo-banner-desktop'}
+                    />
+                </Box>
+            )}
+            {promoBannerMobile && !pageErrorMobile && (
+                <Box display={{base: 'block', md: 'none'}}>
```

Comment:
The `ImageWithText` component changes the text to center align when the browser size is `sm` but the code here shows the mobile image when the browser size is `md`. This is because if we kept the desktop image when the browser size was `md` the text overflows the image.

Reply:
Note that we have similar logic in this component: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/fd7f424dfcafb5233e458b9e93550df58eab0410/packages/template-retail-react-app/app/components/responsive/index.jsx#L30-L31

You may want to use it. Do note though that the breakpoint is at `lg` (not `md`).

Reply:
Yes, I also saw this component and did originally use it but found that the breakpoint being at `lg` made the text overflow the banner and wanted to make the breakpoint at `md`

However, we do have a WI (To be written) to create a new ISML component for the banner that takes the mobile and desktop banner and will handle rendering which banner when in Page Designer, so this logic will be removed then